src/lib/core/data-table/data-table.ts

```diff
@@ -123,12 +157,11 @@ export class CdkTable<T> implements CollectionViewer {
 
     // TODO(andrewseguin): Add trackby function input.
     // Find and construct an iterable differ that can be used to find the diff in an array.
-    this._dataDiffer = this._differs.find([]).create();
+    this._dataDiffer = this._differs.find(this._data).create();
   }
 
   ngOnDestroy() {
-    // TODO(andrewseguin): Disconnect from the data source so
-    //   that it can unsubscribe from its streams.
+    this._onDestroy.next();
```

Comment:
I believe you want to complete the subject so it won't emit anymore and the datasource listener will also finish.

this._onDestroy.complete();